backend/application.py

```diff
@@ -477,3 +480,77 @@ def get(self, dataset):
         self.set_header("Content-Type", logo_entry.mimetype)
         self.write(logo_entry.data)
         self.finish()
+
+
+class SFTPAccess(handlers.AdminHandler):
+    """
+    Creates, or re-enables, sFTP users in the database.
+    """
+    def get(self, dataset):
+        """
+        Returns sFTP credentials for the current user.
+        """
+
+        password = None
+        username = None
+        expires = None
+        # Check if an sFTP user exists for the current user
+        try:
+            data = self.current_user.sftp_user.get()
+            username = data.user_name
+            expires = data.expires.strftime("%Y-%m-%d %H:%M")
+        except db.SFTPUser.DoesNotExist:
+            # Otherwise return empty values
+            pass
+
+        self.finish({'user':username,
+                     'expires':expires,
+                     'password':password})
+
+    def post(self, dataset):
+        """
+        Handles generation of new credentials. This function either creates a
+        new set of sftp credentials for a user, or updates the old ones with a
+        new password and expiry date.
+        """
+
+        # Create a new password
+        username = "_".join(self.current_user.name.split()) + "_sftp"
+        password = self.generate_password()
+        expires = datetime.today() + timedelta(days=30)
+
+        # Check if an sFTP user exists for the current user when the database is ready
+        try:
+            self.current_user.sftp_user.get()
+            # if we have a user, update it
+            db.SFTPUser.update(password_hash = crypt.crypt( password, username ),
+                               account_expires = expires
+                               ).where(db.SFTPUser.user == self.current_user.get_id()).execute()
```

Comment:
You can do this by `db.SFTPUser.user == self.current_user`. No need for the `get_id` part.